fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/ToolManager.java

```diff
@@ -0,0 +1,43 @@
+package net.fabricmc.fabric.api.tool.attribute.v1;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.ActionResult;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.util.TriState;
+import net.fabricmc.fabric.impl.tool.attribute.ToolManagerImpl;
+
+public final class ToolManager {
+	public static Event<ToolHandler> tag(Tag<Item> tag) {
+		return ToolManagerImpl.tag(tag);
+	}
+
+	public static Event<ToolHandler> general() {
+		return ToolManagerImpl.general();
+	}
+
+	public static TriState handleIsEffectiveOn(ItemStack stack, BlockState state, LivingEntity user) {
+		return ToolManagerImpl.handleIsEffectiveOn(stack, state, user);
+	}
+
+	public static float handleBreakingSpeed(ItemStack stack, BlockState state, LivingEntity user) {
+		return ToolManagerImpl.handleBreakingSpeed(stack, state, user);
+	}
+
+	private ToolManager() {
+	}
+
+	public interface ToolHandler {
```

Comment:
I hope this won't mean that every mod needs to register events for all their tools?

Reply:
nope, ideally you don't even need to touch this callback, I have thought of keeping this callback internal, but have decided to leave it open in case someone adds a new tool type that is completely separate from vanilla

Reply:
alright, sounds good.